packages/core/src/wallet/IndyWallet.ts

```diff
@@ -56,13 +56,13 @@ export class IndyWallet implements Wallet {
   }
 
   public get masterSecretId() {
-    if (!this.isInitialized || !this.walletConfig?.id) {
+    if (!this.isInitialized) {
       throw new AriesFrameworkError(
         'Wallet has not been initialized yet. Make sure to await agent.initialize() before using the agent.'
       )
     }
 
-    return this.walletConfig.id
+    return this.walletConfig?.masterSecretId ?? this.walletConfig?.id
```

Comment:
This can lead to undefined being returned, which wasn't possible before. We should probably check if either have a value and throw an error otherwise

Reply:
It's good question. I'm not sure but is it possible case that `isInitialized` is `true` but we don't have `this.walletConfig.id`?
It seems like we have to open wallet to make `isInitialized` to be `true`, but we can't open wallet without wallet id.

Reply:
I think this is the reason why CI is failing (a few lines in `IndyHolderService.ts` are expecting it to be always defined, something that makes sense because otherwise it cannot work).

Reply:
I returned back this check and also aded for `masterSecretId`. Hope now everything will be fine 🙂